runtime/compiler/runtime/ClassUnloadAssumption.cpp

```diff
@@ -379,53 +365,61 @@ void TR_RuntimeAssumptionTable::markForDetachFromRAT(OMR::RuntimeAssumption *ass
  * will be traversed, and only the hashtable linked-lists that have a non-zero 
  * marked for detach count will be traversed.
  */
-void TR_RuntimeAssumptionTable::reclaimMarkedAssumptionsFromRAT()
+void TR_RuntimeAssumptionTable::reclaimMarkedAssumptionsFromRAT(int32_t cleanupCount)
    {
-   TR_RatHT *hashTable;
-   OMR::RuntimeAssumption *cursor, *next, *prev;
-   int kind, reclaimed=0;
+   if (_marked == 0)
+      return;
 
    assumptionTableMutex->enter();
-   for (kind=0; kind < LastAssumptionKind; kind++) // for each table
+   for (int kind=0; _marked > 0 && cleanupCount != 0 && kind < LastAssumptionKind; kind++) // for each table
       {
       if (_detachPending[kind] == true)  // Is there anything to remove from this table?
          {
-         hashTable = _tables + kind;
-         for (size_t i = 0; i < hashTable->_spineArraySize; ++i) // for each bucket in the table
+         TR_RatHT *hashTable = _tables + kind;
```

Comment:
Just an idea: every invocation of `reclaimMarkedAssumptionsFromRAT` could start from a different table (or different bucket) to avoid the situation where the later buckets/tables get larger because they have a lower priority in evacuation.

Reply:
Could I do this as a follow-on change since I worry non-determinism or extra complexity could make debugging harder?

Reply:
Sure.It was a suggestion anyway.